driver-sync/src/main/com/mongodb/client/internal/MongoClientDelegate.java

```diff
@@ -216,12 +225,18 @@ ReadBinding getReadBinding(final ReadPreference readPreference, final ReadConcer
 
         ReadWriteBinding getReadWriteBinding(final ReadPreference readPreference, final ReadConcern readConcern,
                                              @Nullable final ClientSession session, final boolean ownsSession) {
-            ReadWriteBinding readWriteBinding = new ClusterBinding(cluster, getReadPreferenceForBinding(readPreference, session),
-                    readConcern);
+            ClusterAwareReadWriteBinding readWriteBinding = new ClusterBinding(cluster,
+                    getReadPreferenceForBinding(readPreference, session), readConcern);
+
+            if (crypt != null) {
+                readWriteBinding = new CryptBinding(readWriteBinding, crypt);
```

Comment:
This is where everything plugs together: CryptBinding will wrap the ClusterBinding, and ultimately the Connection with CryptConnection, which intercepts all commands and encrypts/decrypts them